service/service.go

```diff
@@ -937,7 +937,7 @@ func (s *service) getCSISnapshot(vol *siotypes.Volume, systemID string) *csi.Sna
 	snapshot := &csi.Snapshot{
 		SizeBytes:      int64(vol.SizeInKb) * bytesInKiB,
 		SnapshotId:     systemID + dash + vol.ID,
-		SourceVolumeId: vol.AncestorVolumeID,
+		SourceVolumeId: systemID + dash + vol.AncestorVolumeID,
```

Comment:
I am also worried about this. Can we use cert-csi to validate snapshot functionality with this change?

Reply:
cert-csi passes. This is a defect that has to be fixed as the source volume ID is incorrect.